src/sentry/static/sentry/app/utils/analytics.tsx

```diff
@@ -182,11 +194,11 @@ metric.measure = function metricMeasure({name, start, end, data = {}, noCleanup}
   const startData = metricDataStore.get(start) || {};
 
   // Retrieve measurement entries
-  performance
-    .getEntriesByName(name, 'measure')
-    .forEach(measurement =>
-      metric(measurement.name, measurement.duration, {...startData, ...data})
-    );
+  performance.getEntriesByName(name, 'measure').forEach(measurement => {
+    if (maxLength === 0 || measurement.duration <= maxLength) {
+      metric(measurement.name, measurement.duration, {...startData, ...data});
+    }
```

Comment:
I might invert this so that we return early if `duration >= maxLength` so that by default we record data (e.g. if maxLength is null).